### PR TITLE
Add support for logging into a file or journald

### PR DESCRIPTION
**Pull request by jonathan-schild**
Supports basic log level.
Option to log and print to stdout.

Not tested under windows yet.

May crash if the user is not allowed to create/open the log file because the exception handling is not yet implemented.
However, this should not be a problem if you use default paths or paths in the user directory.

### CODE REVIEW

File: src/log.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+class LogMode(Enum):
+  SYSTEMD = 0
+  FILE = 1
+  AUTO = 0
+
+class LogLevel(Enum):
+  DEBUG = 0
+  INFO = 1
+  IMPORTANT = 2
```
Comment (Gamer92000) [inline src/log.py:17]: I would prefer DEBUG, INFO, WARNING, ERROR

**Review by Gamer92000 — CHANGES_REQUESTED** [review]
I don't see that logging to stdout will be used, as the stdout is already used for progress updates.

**Comment by jonathan-schild** [comment]
I changed the log level and updated the methods in LogLevel.

> I don't see that logging to stdout will be used, as the stdout is already used for progress updates.
You don't have to use it xD. I thought it could be useful in debugging scenarios though.

**Review by Gamer92000 — APPROVED** [review]
such fancy

**Pull request merged**